src/validation/manual.address.validation.ts

```diff
@@ -0,0 +1,220 @@
+import { Address } from "@companieshouse/api-sdk-node/dist/services/officer-filing";
+import { ManualAddressValidationType, ValidationError } from "../model/validation.model";
+import { COUNTRY_LIST, UK_COUNTRY_LIST } from "../utils/properties";
+
+const REGEX_FOR_MANUAL_ADDRESS = "^[-,.:; 0-9A-Z&@$£¥€'\"«»?!/\\\\()\\[\\]{}<>*=#%+ÀÁÂÃÄÅĀĂĄÆǼÇĆĈĊČÞĎÐÈÉÊËĒĔĖĘĚĜĞĠĢĤĦÌÍÎÏĨĪĬĮİĴĶĹĻĽĿŁÑŃŅŇŊÒÓÔÕÖØŌŎŐǾŒŔŖŘŚŜŞŠŢŤŦÙÚÛÜŨŪŬŮŰŲŴẀẂẄỲÝŶŸŹŻŽa-zſƒǺàáâãäåāăąæǽçćĉċčþďðèéêëēĕėęěĝģğġĥħìíîïĩīĭįĵķĺļľŀłñńņňŋòóôõöøōŏőǿœŕŗřśŝşšţťŧùúûüũūŭůűųŵẁẃẅỳýŷÿźżž]*$";
+const REGEX_FOR_VALID_UK_POSTCODE = /^[A-Z]{1,2}[0-9][A-Z0-9]? ?[0-9][A-Z]{2}$/;
+const PREMISE_ALLOWED_LENGTH = 200;
+const POSTCODE_ALLOWED_LENGTH = 20;
+const ALLOWED_LENGTH = 50;
+
+export const validateManualAddress = (address: Address, manualAddressValidationType: ManualAddressValidationType): ValidationError[] => {
+	let validationErrors: ValidationError[] = [];
+
+	// Premises
+	if(!address.premises) {
+		validationErrors.push(manualAddressValidationType.MissingValue.Premise);
+	} else {
+		validationErrors = validatePremise(address.premises, manualAddressValidationType, validationErrors);
+	}
+
+	// Address Line 1
+	if(!address.addressLine1) {
+		validationErrors.push(manualAddressValidationType.MissingValue.AddressLine1);
+	} else {
+		validationErrors = validateAddressLine1(address.addressLine1, manualAddressValidationType, validationErrors);
+	}
+
+	// Address Line 2
+	if(address.addressLine2) {
+		validationErrors = validateAddressLine2(address.addressLine2, manualAddressValidationType, validationErrors);
+	}
+
+	// City
+	if(!address.locality) {
+		validationErrors.push(manualAddressValidationType.MissingValue.City);
+	} else {
+		validationErrors = validateCity(address.locality, manualAddressValidationType, validationErrors);
+	}
+
+	// County, Town, Province, Region
+	if(address.region) {
```

Comment:
Were you meant to check for the presence or absence of region?

Reply:
no, it's an optional field.